x-pack/plugins/lens/public/indexpattern_plugin/indexpattern.tsx

```diff
@@ -95,76 +124,49 @@ export function IndexPatternDataPanel(props: DatasourceDataPanelProps<IndexPatte
   );
 }
 
-export type IndexPatternDimensionPanelProps = DatasourceDimensionPanelProps & {
-  state: IndexPatternPrivateState;
-  setState: (newState: IndexPatternPrivateState) => void;
-};
+export function columnToOperation(column: IndexPatternColumn) {
+  const { dataType, label, isBucketed, operationId } = column;
+  return {
+    id: operationId,
+    label,
+    dataType,
+    isBucketed,
+  };
+}
 
-export function IndexPatternDimensionPanel(props: IndexPatternDimensionPanelProps) {
-  const fields = props.state.indexPatterns[props.state.currentIndexPatternId].fields;
-  const columns: IndexPatternColumn[] = fields.map((field, index) => ({
-    operationId: `${index}`,
-    label: `Value of ${field.name}`,
-    dataType: field.type as DataType,
-    isBucketed: false,
+type UnwrapPromise<T> = T extends Promise<infer P> ? P : T;
+type InferFromArray<T> = T extends Array<infer P> ? P : T;
 
-    operationType: 'value' as OperationType,
-    sourceField: field.name,
-  }));
+function addRollupInfoToFields(
+  indexPattern: InferFromArray<Exclude<UnwrapPromise<ReturnType<typeof getIndexPatterns>>, void>>
+): IndexPattern {
+  if (!indexPattern.typeMeta || typeof indexPattern.fields === 'string') {
+    return indexPattern;
+  }
 
-  const filteredColumns = columns.filter(col => {
-    const { operationId, label, dataType, isBucketed } = col;
+  const aggs = Object.keys(indexPattern.typeMeta.aggs);
 
-    return props.filterOperations({
-      id: operationId,
-      label,
-      dataType,
-      isBucketed,
+  const newFields = [...(indexPattern.fields as IndexPatternField[])];
+  newFields.forEach((field, index) => {
+    const restrictionsObj: IndexPatternField['rollupRestrictions'] = {};
```

Comment:
nit: We could give this object its own type.

Reply:
I'm not a big fan of this suggestion because I was already struggling to organize the types here. I will not make this change unless we have another use case for it. I will rename this from rollupRestrictions to aggregationRestrictions